File: server/repositories/conversationRepository.ts
import { getDb } from '../db.js';
import { ConversationRow, Conversation } from '../types.js';

// 数据库 snake_case → API camelCase 转换
function toCamelCase(row: ConversationRow): Conversation {
  return {
    id: row.id,
    title: row.title,
    type: row.type,
    createdAt: row.created_at,
    updatedAt: row.updated_at,
    lockedAgent: row.locked_agent || null,
    routingMode: row.routing_mode || 'auto',
  };
}

// 获取所有会话，按更新时间降序，可选按 type 过滤
export function findAll(type?: string): Conversation[] {
  const db = getDb();
  let sql = 'SELECT id, title, type, created_at, updated_at, locked_agent, routing_mode FROM conversations';
  const params: string[] = [];
  if (type) {
    sql += ' WHERE type = ?';
    params.push(type);
  }
  sql += ' ORDER BY updated_at DESC';
  const rows = db.prepare(sql).all(...params) as ConversationRow[];
  return rows.map(toCamelCase);
}

export function findById(id: string): Conversation | null {
  const db = getDb();
  const row = db.prepare(
    'SELECT id, title, type, created_at, updated_at, locked_agent, routing_mode FROM conversations WHERE id = ?'
  ).get(id) as ConversationRow | undefined;
  return row ? toCamelCase(row) : null;
}

export function create({ id, title, type, routingMode }: { id: string; title: string; type?: string; routingMode?: string }): Conversation {
  const db = getDb();
  const now = new Date().toISOString();
  const convType = type || 'text';
  const mode = routingMode || 'auto';
  db.prepare(
    'INSERT INTO conversations (id, title, type, created_at, updated_at, locked_agent, routing_mode) VALUES (?, ?, ?, ?, ?, ?, ?)'
  ).run(id, title, convType, now, now, null, mode);
  return { id, title, type: convType, createdAt: now, updatedAt: now, lockedAgent: null, routingMode: mode };
}

// 删除会话，返回受影响行数（调用方据此判断是否 404）
export function deleteById(id: string): { changes: number } {
  const db = getDb();
  return db.prepare('DELETE FROM conversations WHERE id = ?').run(id);
}

export function updateTitle(id: string, title: string): Conversation | null {
  const db = getDb();
  const now = new Date().toISOString();
  const result = db.prepare(
    'UPDATE conversations SET title = ?, updated_at = ? WHERE id = ?'
  ).run(title, now, id);
  if (result.changes === 0) return null;
  return findById(id);
}

// 更新锁定 Agent（为 null 时解锁）
export function updateLockedAgent(id: string, lockedAgent: string | null): Conversation | null {
  const db = getDb();
  const now = new Date().toISOString();
  const result = db.prepare(
    'UPDATE conversations SET locked_agent = ?, updated_at = ? WHERE id = ?'
  ).run(lockedAgent, now, id);
  if (result.changes === 0) return null;
  return findById(id);
}

File: server-py/repositories/conversation_repository.py
from __future__ import annotations

import sqlite3
from typing import Any, Optional

from database import get_db, now


def _row_to_dict(row: sqlite3.Row) -> dict[str, Any]:
    """将 SQLite Row 转换为驼峰命名的字典（匹配前端期望的 JSON 格式）"""
    return {
        "id": row["id"],
        "title": row["title"],
        "createdAt": row["created_at"],
        "updatedAt": row["updated_at"],
    }


def find_all() -> list[dict[str, Any]]:
    """获取所有对话，按 updated_at 降序排列（最新更新的在前）"""
    db = get_db()
    rows = db.execute(
        "SELECT id, title, created_at, updated_at FROM conversations ORDER BY updated_at DESC"
    ).fetchall()
    return [_row_to_dict(r) for r in rows]


def find_by_id(id: str) -> Optional[dict[str, Any]]:
    db = get_db()
    row = db.execute(
        "SELECT id, title, created_at, updated_at FROM conversations WHERE id = ?", (id,)
    ).fetchone()
    return _row_to_dict(row) if row else None


def create(id: str, title: str) -> dict[str, Any]:
    db = get_db()
    ts = now()
    db.execute(
        "INSERT INTO conversations (id, title, created_at, updated_at) VALUES (?, ?, ?, ?)",
        (id, title, ts, ts),
    )
    return find_by_id(id)


def delete_by_id(id: str) -> int:
    """删除对话，返回受影响行数（0 表示 ID 不存在）"""
    db = get_db()
    cursor = db.execute("DELETE FROM conversations WHERE id = ?", (id,))
    return cursor.rowcount


def update_title(id: str, title: str) -> Optional[dict[str, Any]]:
    """更新对话标题并刷新 updated_at 时间戳"""
    db = get_db()
    cursor = db.execute(
        "UPDATE conversations SET title = ?, updated_at = ? WHERE id = ?",
        (title, now(), id),
    )
    if cursor.rowcount == 0:
        return None
    return find_by_id(id)
